### PR TITLE
Changed FeatureAlgorithmFactory.cpp to allow pathnames in the algorithm parameter. (Adresses issue #2474)

### PR DESCRIPTION
**Pull request by paarongiroux**
## Description

Added a regular expression in the formatSpecifications method to check if the savepath parameter is present. If so, the method will only split on "/" characters before this parameter. Otherwise it was split on "/" as it was previously doing.
## Related Issue

https://github.com/USGS-Astrogeology/ISIS3/issues/2474 
## Motivation and Context

Part of support sprint.

## How Has This Been Tested?

Ran the application tests to make sure everything still works, ran the application with a command that previously broke it which now works.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jlaura** [comment]
Checkout: https://www.internalpointers.com/post/squash-commits-into-one-git

**Comment by jessemapel** [comment]
@jlaura Why are you concerned about squashing these down? They will be squashed on merge anyway.

**Comment by jlaura** [comment]
So that we can get a clean commit message in case we need to back them back out. Right now I believe the message will be the final commit message.

File: isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp [hunk 1]
```diff
@@ -511,8 +511,14 @@ SharedRobustMatcher FeatureAlgorithmFactory::make(const QString &definition) con
  *      createMatcher(extractor))..
  */
 QStringList FeatureAlgorithmFactory::formatSpecifications(QString specification) const {
-  QStringList parts = specification.split("/", QString::SkipEmptyParts);
-
+  QStringList parts;
+  if ( specification.contains(QRegularExpression("@savepath", QRegularExpression::CaseInsensitiveOption)) ) {
+    QRegularExpression sep("/(?=(.*(@savepath)))",QRegularExpression::CaseInsensitiveOption);
+    parts = specification.split(sep, QString::SkipEmptyParts);
```
Comment (SgStapleton) [inline isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:517]: I just want to confirm: if you are splitting on a regex that includes "@savepath", does this not mean that the parts list would now be missing "@savepath" in it's elements...? I might just be misunderstanding the regex.

Comment (SgStapleton) [reply isis/src/control/apps/findfeatures/FeatureAlgorithmFactory.cpp:517]: Confirmed with @paarongiroux that this regex does not strip the "@savepath" from the parts list.

**Comment by SgStapleton** [comment]
Good job overall, guys! Don't forget to add a history comment to the FeatureAlgorithmFactory.h file, and this should be good to go!

**Comment by paarongiroux** [comment]
@SgStapleton I added the history comment so everything should be good now.

**Review by SgStapleton — APPROVED** [review]
This looks good to go!

**Pull request merged**